client/renderSDL/SDL_Extensions.cpp

```diff
@@ -659,16 +664,16 @@ void CSDL_Ext::convertToGrayscale( SDL_Surface * surf, const Rect & rect )
 template<int bpp>
 void scaleSurfaceFastInternal(SDL_Surface *surf, SDL_Surface *ret)
 {
-	const float factorX = float(surf->w) / float(ret->w),
-				factorY = float(surf->h) / float(ret->h);
+	const float factorX = float(surf->w) / float(ret->w);
+	const float factorY = float(surf->h) / float(ret->h);
 
 	for(int y = 0; y < ret->h; y++)
 	{
 		for(int x = 0; x < ret->w; x++)
 		{
 			//coordinates we want to calculate
-			int origX = static_cast<int>(floor(factorX * x)),
-				origY = static_cast<int>(floor(factorY * y));
+			auto origX = static_cast<int>(floor(factorX * x));
```

Comment:
Changed `int` to `auto` to address this:

> 
> Replace the redundant type with "auto".
> "auto" should be used to avoid repetition of types [cpp:S5827](https://sonarcloud.io/organizations/alexander-wilms/rules?open=cpp%3AS5827&rule_key=cpp%3AS5827)